ghcide/src/Development/IDE/Types/Options.hs

```diff
@@ -137,8 +130,16 @@ defaultIdeOptions session = IdeOptions
     ,optCheckParents = pure CheckOnSaveAndClose
     ,optHaddockParse = HaddockParse
     ,optCustomDynFlags = id
+    ,optSkipProgress = defaultSkipProgress
     }
 
+defaultSkipProgress :: Typeable a => a -> Bool
+defaultSkipProgress key = case () of
```

Comment:
Could we comment as to why these nodes get excluded? The original reason for excluding GetFileExists was that given a module `Foo.hs`, we might have N `GetFileExists` at each include path, at `.lhs`, at `.hs-boot` etc. Whereas every other rule is pretty much only at a single filepath. Is that property true for modtime? It definitely isn't for getfilecontents, but is that about performance?

Reply:
It's performance, but I haven't measured. I'll add a comment or revert to just `GetFileExists`

Reply:
I don't think is unreasonable - its a commonly called thing that is mostly irrelevant for performance concerns. I do think its important to say that `GetFileExists` is specially, because its not obvious that is has that weird behaviour, and how if you do include `GetFileExists` your progress is basically meaningless.